src/gt4py/analysis/passes.py

```diff
@@ -700,6 +706,28 @@ def visit_TernaryOpExpr(self, node: gt_ir.TernaryOpExpr, **kwargs):
                 node.then_expr.data_type, node.else_expr.data_type
             )
 
+        def visit_NativeFuncCall(self, node: gt_ir.NativeFuncCall, **kwargs):
+            self.generic_visit(node.args, **kwargs)
+
+            # Collect data types from each argument
+            data_type = set([arg.data_type for arg in node.args])
+            if len(data_type) > 1:
+                data_type -= {gt_ir.DataType.DEFAULT}
+
+            # Assert that all data types are the same
+            assert len(data_type) == 1
+            data_type = data_type.pop()
```

Comment:
code suggestion implemented.

As for the discussion:
We think in the long run the type promotion should be what we aim for (since for these operators, python only requires a `__ge__` operator) but we do not think this should be handled within the scope of this. To track this we will add an issue if needed?

Reply:
Ok, I agree with you. It would be great if you could open an issue to track the type promotion discussion in the future.